ogr/abstract.py

```diff
@@ -1394,3 +1394,20 @@ def get_projects(self) -> Sequence["GitProject"]:
 
     def get_forks(self) -> Sequence["GitProject"]:
         raise NotImplementedError
+
+
+class GithubTokenManager:
+    """
+    Represents a token manager for authentication via GitHubApp.
```

Comment:
I would place a note about remote service we need to run for that.

```suggestion
    Represents a token manager for authentication via GitHubApp using a remote REST service.
```

Reply:
Not necessarily, the default token manager (when you provide _GitHub App ID_ and key) doesn't use REST and utilizes `PyGithub` directly.

Reply:
OK, now I get the architecture.

Reply:
What about renaming the class to  `GithubAppTokenManager`?

Reply:
> What about renaming the class to `GithubAppTokenManager`?

The way `PyGithub` instances are retrieved, you could create a subclass that uses personal access token**s**. Even the default token auth that is included in `GithubService` could be extracted into `TokenManager` subclass. (That would be probably a mess to handle in `__init__` (since it's passed by raw data, e.g. token), but it would remove all the logic regarding auth from `GithubService`)

(Also the subclasses already look like Java classes because of the naming) :/

Reply:
That was my point that you use `TokenManager` only for Github apps so it makes sense to name it like that.

Will it be so hard to handle token auth in the token manager subclass to have it consistent and exclude the authentication completely?

Reply:
The main problem is in service constructor, since it takes info of each auth method in separate arguments, that is what decides if token or token manager (either Tokman or github app through pygithub) is used and it's done in service.
(Which also makes checking for multiple provided auth methods interesting (passing tuple to check if only one is not None))

Separate classes for each auth method would provide nice encapsulation, but I wonder how much code is tied to public attributes related to that. Creating those objects probably from factory-like method, like is used for parsing services?

Reply:
> The main problem is in service constructor

Yes, I got that but still think we can try to separate that...;)

>  I wonder how much code is tied to public attributes related to that.

I am not sure but probably mostly tests. The functionality should not use that.

> Creating those objects probably from factory-like method, like is used for parsing services?

:+1: